src/exa.h

```diff
@@ -37,17 +37,40 @@ typedef struct tegra_attrib_bo {
     __fp16 *map;
 } TegraEXAAttribBo;
 
+enum Tegra2DRotation {
+    TEGRA2D_FLIP_X,
+    TEGRA2D_FLIP_Y,
+    TEGRA2D_TRANS_LR,
+    TEGRA2D_TRANS_RL,
+    TEGRA2D_ROT_90,
+    TEGRA2D_ROT_180,
+    TEGRA2D_ROT_270,
+    TEGRA2D_IDENTITY,
```

Comment:
This seems a bit clunky. Isn't it cleaner to have a bitflag that can encode every possible orientation/flip variation?

Something like this has [served me well](https://github.com/fuse-open/fuselibs/blob/master/Source/Fuse.Elements/Resources/Exif/Exif.uno#L12) in the past:
```c
enum TegraOrientation {
	TEGRA2D_IDENTITY = 0,
	TEGRA2D_ROT_90 = 1 << 0,
	TEGRA2D_ROT_180 = 1 << 1,
	TEGRA2D_FLIP_Y = 1 << 2,
}
```
All other transformations should be possible to be expressed in terms of combinations of these (FLIP_Y can be traded for FLIP_X, it's not really significant).

I also think "orientation" is a clearer term for this than "rotation".

Reply:
Those enum's are matching the values that hardware uses and we're applying them directly to the HW register value. GR2D can't combine rotation with flipping, hence I don't see it worthy to use the bitflags.

Reply:
Yes, "orientation" indeed sounds a bit more appropriate, thanks.

Reply:
Uh, does the HW interface really enumerate all possible orientations separately? 

```
    TEGRA2D_FLIP_X   = 0 = 0b000,
    TEGRA2D_FLIP_Y   = 1 = 0b001,
    TEGRA2D_TRANS_LR = 2 = 0b010,
    TEGRA2D_TRANS_RL = 3 = 0b011,
    TEGRA2D_ROT_90   = 4 = 0b100,
    TEGRA2D_ROT_180  = 5 = 0b101,
    TEGRA2D_ROT_270  = 6 = 0b110,
    TEGRA2D_IDENTITY = 7 = 0b111,
```
There's *no* pattern here... yikes! That's a pretty bonkers design, but OK :laughing:

Reply:
Nothing is ideal :)

Reply:
:cry: